### PR TITLE
WT-12110 Disable timestamp_abort backup tests in the compatibility mode

### PR DESCRIPTION
**Pull request by pmacko86**

### CODE REVIEW

**Comment by pmacko86** [comment]
Most importantly, we need to make sure the test doesn't run, so we should definitely turn it off at the top level as in the PR.

I didn't think it was necessary, but if we'd like we could also add testutil_assert to check for certain combinations of test options?

File: test/csuite/timestamp_abort/smoke.sh [hunk 1]
```diff
@@ -33,6 +33,9 @@ $TEST_WRAPPER $test_bin -m $default_test_args -c
 #$TEST_WRAPPER $test_bin -m $default_test_args -L
 $TEST_WRAPPER $test_bin -C $default_test_args
 $TEST_WRAPPER $test_bin -C $default_test_args -c
-$TEST_WRAPPER $test_bin -C $default_test_args -B -I 3
 $TEST_WRAPPER $test_bin -C -m $default_test_args
 $TEST_WRAPPER $test_bin -C -m $default_test_args -c
+
+# Don't test backups in the compatibility mode. MongoDB no longer uses it, and the compatibility
+# mode is inherently incompatible with backup-related fixes that add new log records.
+#$TEST_WRAPPER $test_bin -C $default_test_args -B -I 3
```
Comment (sueloverso) [inline test/csuite/timestamp_abort/smoke.sh:41]: I agree this is the first item to change and it solves the testing problem immediately.

**Review by sueloverso — APPROVED** [review]
LGTM

**Comment by sueloverso** [comment]
I agree with your change. I think adding it to the code too would be good. If someone ever uncomments that line then they'd at least get an error instead of a difficult to debug issue first. If it ever is supposed to work, then the error conditional could be removed. But it feels safer to do both here.

**Review by sueloverso — APPROVED** [review]
LGTM Thanks!

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.

**Comment by pmacko86** [comment]
evergreen merge

**Pull request merged**